### DISCUSSION TITLE
Property 'rateLimit' does not exist on type 'Request<ParamsDictionary, any, any, ParsedQs, Record<string, any>>'

### DISCUSSION THREAD

**Discussion by SaidiAli** [nfriedly/express-rate-limit#454] [Q&A] [answered]
Am getting this error when I try to access the rateLimit property on the express request object. I want to use this to confirm that the limiting works. How do I make this work?

```
app.get('/ip', (req: Request, res: Response) => {
    res.json({ ip: req.ip, limiter: req.rateLimit.remaining });
  }); // this is for checking if server is behind a proxy/load balancer
```

**Comment by gamemaker1** [comment]
You could use the following:

```ts
import type { AugmentedRequest } from 'express-rate-limit'

app.get('/ip', (req: AugmentedRequest, res: Response) => {
    res.json({ ip: req.ip, limiter: req.rateLimit.remaining });
  }); // this is for checking if server is behind a proxy/load balancer
```

**Reply by SaidiAli** [reply]
Thank you for the quick response. Currently testing it out

**Comment by SaidiAli** [comment]
Hello @gamemaker1 , thank you for your answer it worked.
I also wanted to ask if there is a way to set the reset time. I have checked the docs and there is nothing in configuration for setting reset time.

**Reply by gamemaker1** [reply, marked as answer]
The reset time is set by the middleware itself, you don't need to set it yourself.

You can access it in all requests handlers registered after the rate limiter using `req.rateLimit`.

**Reply by nfriedly** [reply]
The `windowMs` configuration option controls the reset time: https://express-rate-limit.mintlify.app/reference/configuration#windowms

I'll add something to the description to clarify that.

**Reply by SaidiAli** [reply]
Thanks guys.
